tests/test_kafka.py

```diff
@@ -150,21 +67,9 @@ def test_pipeline(kerberos_flag, stop_count, jar_uri, keytab_secret, spark_servi
         "--conf", uris
     ]
 
-    kerberos_args = [
-        "--conf", "spark.mesos.driver.secret.names={}".format(keytab_secret),
-        "--conf", "spark.mesos.driver.secret.filenames=kafka-client.keytab",
-        "--conf", "spark.mesos.executor.secret.names={}".format(keytab_secret),
-        "--conf", "spark.mesos.executor.secret.filenames=kafka-client.keytab",
-        "--conf", "spark.mesos.task.labels=DCOS_SPACE:/{}".format(spark_service_name),
-        "--conf", "spark.executorEnv.KRB5_CONFIG_BASE64={}".format(KAFKA_KRB5),
-        "--conf", "spark.mesos.driverEnv.KRB5_CONFIG_BASE64={}".format(KAFKA_KRB5),
-        "--conf", "spark.driver.extraJavaOptions=-Djava.security.auth.login.config="
-                  "/mnt/mesos/sandbox/spark-kafka-client-jaas.conf",
-        "--conf", "spark.executor.extraJavaOptions="
-                  "-Djava.security.auth.login.config=/mnt/mesos/sandbox/spark-kafka-client-jaas.conf",
-    ]
+    kerberos_args = get_kerberized_kafka_spark_conf(spark_service_name)
 
-    producer_config = ["--conf", "spark.cores.max=2", "--conf", "spark.executor.cores=2",
+    producer_config = ["--conf", "spark.cores.max=2", "--conf", "spark.executor.cores=1",
```

Comment:
This will mean that we spawn 2 producers, is this intended?

Reply:
yes, the main goal here is to lower down a single executor resource demands while keeping parallelism at the same level.